tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerInvoker.cs

```diff
@@ -46,14 +47,25 @@ public static MethodDebuggerState BeginMethod_StartMarker<TTarget>(string probeI
                 return CreateInvalidatedDebuggerState();
             }
 
-            var state = new MethodDebuggerState(probeId, scope: default, DateTimeOffset.UtcNow, methodMetadataIndex, instance);
-            state.SnapshotCreator.StartDebugger();
+            var state = new MethodDebuggerState(probeId, scope: default, DateTimeOffset.UtcNow, methodMetadataIndex, instance, new ProbeCondition(" ", MethodPhase.Entry, " "));
```

Comment:
TODO - fill in these empty strings...

Reply:
This is currently allocation a `ProbeCondition` object on the heap on each execution, which would drive the GC crazy if the method is executed often.